decidim-proposals/spec/features/vote_proposal_spec.rb

```diff
@@ -69,7 +71,59 @@
           end
 
           within "#proposal-#{proposal.id}-votes-count" do
-            expect(page).to have_content("1 VOTE")          
+            expect(page).to have_content("1 VOTE")
+          end
+        end
+      end
+
+      context "when the feature has a vote limit" do
+        before do
+          feature.settings = {
+            vote_limit: 10
+          }
+          feature.save
+          visit_feature
+        end
+
+        context "when the proposal is not voted yet" do
+          it "should update the remaining votes counter" do
+            within "#proposal-#{proposal.id}-vote-button" do
+              page.find('.card__button').click
+              expect(page).to have_css('.card__button.success')
+            end
+
+            expect(page).to have_content("REMAINING 9 VOTES")
+          end
+        end
+
+        context "when the proposal is already voted" do
+          before do
+            create(:proposal_vote, proposal: proposal, author: user)
+            visit_feature
+          end
+
+          it "should be able to undo the vote" do
+            within "#proposal-#{proposal.id}-vote-button" do
+              expect(page).to have_css('.card__button.success')
+              page.find('.card__button').click
+            end
+
+            expect(page).to have_content("REMAINING 10 VOTES")
+          end
+        end
+
+         context "when the user has reached the votes limit" do
+          before do
+            feature.settings = {
```

Comment:
can we move this to a trait or a fake attribute at a factory that sets the settings?